File: starters/Go/bomberjam/Game.go
package bomberjam

import (
    "bufio"
    "encoding/json"
    "errors"
    "fmt"
    "os"
    "strconv"
    "strings"
)

type Game struct {
    stdin      *bufio.Reader
    isReady    bool
    MyPlayerId string
    MyPlayer   *Player
    State      *State
}

func NewGame() Game {
    return Game{
        stdin:      bufio.NewReader(os.Stdin),
        MyPlayerId: "",
        MyPlayer:   nil,
        State:      nil,
    }
}

func (g *Game) writeLine(text string) {
    fmt.Println(text)
}

func (g *Game) readLine() string {
    text, _ := g.stdin.ReadString('\n')
    return strings.TrimSpace(text)
}

func (g *Game) Ready(playerName string) error {
    if g.isReady {
        return nil
    }

    if len(strings.TrimSpace(playerName)) == 0 {
        return errors.New("your name cannot be null or empty")
    }

    g.writeLine("0:" + playerName)
    myPlayerId := g.readLine()

    if _, err := strconv.Atoi(myPlayerId); err != nil {
        return errors.New("could not retrieve your ID from standard input: '" + myPlayerId + "'")
    }

    g.MyPlayerId = myPlayerId
    g.isReady = true
    return nil
}

func (g *Game) ReceiveCurrentState() error {
    var state State
    if err := json.Unmarshal([]byte(g.readLine()), &state); err != nil {
        return err
    }

    g.State = &state
    myPlayer := state.Players[g.MyPlayerId]
    g.MyPlayer = &myPlayer
    return nil
}

func (g *Game) SendAction(action ActionKind) {
    tickStr := strconv.Itoa(g.State.Tick)
    g.writeLine(tickStr + ":" + string(action))
}

File: starters/PHP/bomberjam/Game.php
<?php

namespace bomberjam;

use Exception;

class Game
{
    /** @var bool */
    private $isReady;

    /** @var string */
    public $myPlayerId;

    /** @var Player */
    public $myPlayer;

    /** @var State */
    public $state;

    public function __construct()
    {
        $this->isReady = false;
        $this->myPlayerId = null;
        $this->state = null;
    }

    /**
     * @param $playerName string
     * @throws Exception
     */
    public function ready($playerName)
    {
        if ($this->isReady)
            return;

        if (!isset($playerName) || trim($playerName) === '')
            throw new Exception('Your name cannot be null or empty');

        $this->stdout('0:' . $playerName);

        if (($this->myPlayerId = $this->stdin()) === false || !is_numeric($this->myPlayerId))
            throw new Exception('Could not retrieve your ID from standard input');

        $this->isReady = true;
    }

    /**
     * @throws Exception
     */
    public function receiveCurrentState()
    {
        $jsonStateStr = $this->stdin();
        if ($jsonStateStr === false)
            throw new Exception('Could not retrieve the current state from standard input');

        $jsonState = json_decode($jsonStateStr, true, 512, JSON_THROW_ON_ERROR);
        $this->state = new State($jsonState);

        $this->myPlayer = $this->state->players[$this->myPlayerId];
    }

    /**
     * @param $action string
     */
    public function sendAction($action)
    {
        $this->stdout($this->state->tick . ':' . $action);
    }

    private function stdin()
    {
        return (($text = fgets(STDIN)) === false) ? false : rtrim($text);
    }

    private function stdout($text)
    {
        fwrite(STDOUT, $text . PHP_EOL);
    }
}